src/core/lib/slice/slice_buffer.cc

```diff
@@ -262,54 +276,79 @@ void grpc_slice_buffer_move_into(grpc_slice_buffer* src,
   src->length = 0;
 }
 
+template <bool incref>
+static void slice_buffer_move_single_slice(grpc_slice_buffer* src,
+                                           grpc_slice_buffer* dst,
+                                           size_t bytes_extra) {
+  // Last slice in batch for slice_buffer_move_first_maybe_ref.
+  // We may have to consume it  whole, or just a chunk.
+  if (bytes_extra) {
+    // Just a chunk.
+    if (incref) {
+      dst->slices[dst->count] = grpc_slice_split_head(
+          &src->slices[0], GRPC_SLICE_LENGTH(src->slices[0]) - bytes_extra);
+    } else {
+      dst->slices[dst->count] = grpc_slice_split_head_noref(
+          &src->slices[0], GRPC_SLICE_LENGTH(src->slices[0]) - bytes_extra);
+    }
+  } else {
+    // The whole thing.
+    dst->slices[dst->count] = src->slices[0];
+    ++src->slices;
+    --src->count;
+  }
+  ++dst->count;
+}
+
+template <bool incref>
 static void slice_buffer_move_first_maybe_ref(grpc_slice_buffer* src, size_t n,
-                                              grpc_slice_buffer* dst,
-                                              bool incref) {
-  GPR_ASSERT(src->length >= n);
-  if (src->length == n) {
+                                              grpc_slice_buffer* dst) {
+  if (n == 0) {
+    return;
+  }
+  if (n == src->length) {
```

Comment:
This looks like a Yoda comparison even though it technically isn't; any reason for the change?

Reply:
I agree that the zero case should come first

Reply:
To clarify: do you wish me to make a change here, and specifically what change?

Reply:
My request was to flip the comparison to `if (src->length == n)` as it used to be